### ISSUE TITLE
The ability to store SEO survey data related to SEO assets

### ISSUE THREAD

**Issue by officite** [ahuensch/officite#44]
Add a completed and date field on the SEO asset for completed surverys. There are currently two different ones sent to clients:
SEO Business Goals Survey
SEO Client Satisfaction Survey
Maybe create a drop down for survey type and date completed? Whatever you think works best.

**Comment by officite** [comment]
Also, is it possible to create a report for completed surveys and type as well? I want to be able to keep track or frequency of survey completion. See Lindsey with questions.

**Comment by ahuensch** [comment]
Can a client complete the same survey more than once?

On Mon, Jun 29, 2015 at 10:59 AM, officite <notifications@github.com> wrote:

> Also, is it possible to create a report for completed surveys and type as
> well? I want to be able to keep track or frequency of survey completion.
> See Lindsey with questions.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ahuensch/officite/issues/44#issuecomment-116741626>.
>



-- 
*Anna Huensch*
1546 North LaSalle, Chicago, IL 60610
t *630.639.1960* |  f *866.724.9913*

**Comment by ahuensch** [comment]
Is there a URL or surveymonkey ID that we could store to map back to the
results?

Basically, where I am going with this... I think it might be a good idea to
setup a related record to the asset for survey results.  A search
specialist would click add survey results and then select the date, which
survey, and link it back to survey monkey somehow.  Then you could see a
list of all the survey's completed by that client related to that product.
It will be easier to report off of and it will setup better groundwork for
making sure we're not overwriting data.

 What do you think?

On Mon, Jun 29, 2015 at 11:07 AM, Anna Huensch <anna.huensch@gmail.com>
wrote:

> Can a client complete the same survey more than once?
>
> On Mon, Jun 29, 2015 at 10:59 AM, officite <notifications@github.com>
> wrote:
>
>> Also, is it possible to create a report for completed surveys and type as
>> well? I want to be able to keep track or frequency of survey completion.
>> See Lindsey with questions.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/ahuensch/officite/issues/44#issuecomment-116741626>.
>>
>
>
>
> --
> *Anna Huensch*
> 1546 North LaSalle, Chicago, IL 60610
> t *630.639.1960* |  f *866.724.9913*
>



-- 
*Anna Huensch*
1546 North LaSalle, Chicago, IL 60610
t *630.639.1960* |  f *866.724.9913*

**Comment by ahuensch** [comment]
I added the new features to support storing SEO survey results this weekend. 

As survey results are returned, search specialists will now be able to log the each survey to the appropriate SEO asset. 

**Note** I briefly looked into surveymonkey-to-salesforce integration, but it looks too expensive for our current usage (i.e. in the thousands to implement) but definitely something to keep in mind

Summary - 
Added custom field to asset - # Completed Surveys (roll up) - counts the number of survey results logged on this particular asset
Added custom field to asset - Last Completed Survey Date (roll up) - rolls up the most recent Completed Date from all related Survey Results on this particular asset
Added custom object Survey Result - for storing information about each survey returned related to that asset
Added custom field to survey result - Survey (picklist) - indicates the title of the survey completed
Added custom field to survey result - Date Completed (date) - indicates the date the survey was completed by client
Added Notes & Attachments related list to survey results - for storing survey exports from surveymonkey.com

Attached is a quick how-to for intended design for adding a survey result to an asset.

Below is a link to a report of all survey results submitted:

SEO Survey Results
https://na13.salesforce.com/00Oa0000008nNJE
